libs/librepcb/library/pkg/packagepadlistmodel.cpp

```diff
@@ -88,6 +88,17 @@ void PackagePadListModel::addPad(const QVariant& editData) noexcept {
   try {
     QScopedPointer<UndoCommandGroup> cmd(
         new UndoCommandGroup(tr("Add package pad(s)")));
+    // if no name is set we search for the next free numerical pad name
+    if (mNewName.isEmpty()) {
+      for (unsigned i = 1; i < 32; ++i) {
```

Comment:
This obviously only works for pad names up to 31. Is there a better solution for this problem?

Reply:
Up until 4a03d8491b7421af79935fdd7216b4c2fdd2494a it was done like this:
```cpp
QString PackagePadListEditorWidget::getNextPadNameProposal() const noexcept {
  int i = 1;
  while (mPadList->contains(QString::number(i))) {
    ++i;
  }
  return QString::number(i);
}
```

Reply:
Seems that this is actually resolved :)

Reply:
I actually was a bit hesistant to solve it that way, because it may be `O(n²)`. But I tested it with 128 pads and it wasn't really noticeable.